src/church/lists.rs

```diff
@@ -474,6 +474,43 @@ pub fn filter() -> Term {
     )
 }
 
+/// Applied to a Church-encoded list it returns the last element.
+///
+/// LAST := Z (λzl.NULL l (λx.NIL) (λx.NULL (tail l) (head l) (z (tail l))) I) =
+/// Z (λ 2 1. NULL 1 (λ NIL) (λ NULL (tail 2) (head 2) (3 (tail 2))) I)
```

Comment:
Can you change the names of head and tail to HEAD and TAIL? It's the convention in other docs.